internal/services/springcloud/spring_cloud_customized_accelerator_resource.go

```diff
@@ -396,88 +400,78 @@ func (s SpringCloudCustomizedAcceleratorResource) Delete() sdk.ResourceFunc {
 	return sdk.ResourceFunc{
 		Timeout: 30 * time.Minute,
 		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
-			client := metadata.Client.AppPlatform.CustomizedAcceleratorClient
+			client := metadata.Client.AppPlatform.AppPlatformClient
 
-			id, err := parse.SpringCloudCustomizedAcceleratorID(metadata.ResourceData.Id())
+			id, err := appplatform.ParseCustomizedAcceleratorID(metadata.ResourceData.Id())
 			if err != nil {
 				return err
 			}
 
-			future, err := client.Delete(ctx, id.ResourceGroup, id.SpringName, id.ApplicationAcceleratorName, id.CustomizedAcceleratorName)
+			err = client.CustomizedAcceleratorsDeleteThenPoll(ctx, *id)
 			if err != nil {
 				return fmt.Errorf("deleting %s: %+v", *id, err)
 			}
 
-			if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-				if !response.WasNotFound(future.Response()) {
-					return fmt.Errorf("waiting for deletion of %s: %+v", *id, err)
-				}
-			}
-
 			return nil
 		},
 	}
 }
 
-func expandSpringCloudCustomizedAcceleratorGitRepository(repository []GitRepositoryModel) *appplatform.AcceleratorGitRepository {
+func expandSpringCloudCustomizedAcceleratorGitRepository(repository []GitRepositoryModel) appplatform.AcceleratorGitRepository {
 	if len(repository) == 0 {
-		return nil
+		return appplatform.AcceleratorGitRepository{}
```

Comment:
Will this set an empty json object in the request body when the repository is not specified?

Reply:
This is a required block and has required nested field, actually this if-statement won't be executed.